pkg/hostpath/hostpath.go

```diff
@@ -50,6 +50,8 @@ type hostPath struct {
 	ephemeral         bool
 	maxVolumesPerNode int64
 
+	testSoNoMounter bool
```

Comment:
Shouldn't we mock/stub this interface instead of adding this parameter especially to a public method?
Also, by reading the tests comments, it seems that we will be needing to have sudo privilege to run the tests; which is a bit complex.

Reply:
fyi k8s over at https://github.com/kubernetes/utils/tree/master/mount does not have a "fake" mounter

Reply:
and yes with all this we are "avoiding" the need for sudo priv's that are required to call the "real" mounter

Reply:
I am glad to see that now we have something more elegant than k8s folks :-D